cmd/main.go

```diff
@@ -421,47 +424,51 @@ Full docs can be found at github.com/aws/amazon-` + binName
 		sortField = &sortFieldShorthandPath
 	}
 
-	outputFn := getOutputFn(outputFlag, selector.InstanceTypesOutputFn(resultsOutputFn))
-	var instanceTypes []string
-	var itemsTruncated int
+	// fetch instance types without truncating results
+	prevMaxResults := filters.MaxResults
+	filters.MaxResults = nil
+	instanceTypesDetails, err := instanceSelector.FilterVerbose(filters)
+	if err != nil {
+		fmt.Printf("An error occurred when filtering instance types: %v", err)
+		os.Exit(1)
+	}
 
+	// filtering already sorts by name in ascending order, so only sort if another sort field or
+	// direction is selected
 	sortDirection := cli.StringMe(flags[sortDirection])
-	if *sortField == instanceNamePath && (*sortDirection == sortAscending || *sortDirection == sortAsc) {
-		// filter already sorts in ascending order by name
-		instanceTypes, itemsTruncated, err = instanceSelector.FilterWithOutput(filters, outputFn)
+	if *sortField != instanceNamePath || (*sortDirection != sortAscending && *sortDirection != sortAsc) {
```

Comment:
sorting an already sorted list will be super fast, so I would remove the condition to skip sorting to keep things simpler.  The less if branches the better :)

Reply:
That is a very good point. I will remove the if statement and sort in all cases.